frappe/utils/__init__.py

```diff
@@ -529,7 +529,10 @@ def get_request_site_address(full_address=False):
 
 
 def get_site_url(site):
-	return f"http://{site}:{frappe.get_conf(site).webserver_port}"
+	conf = frappe.get_conf(site)
+	if conf.host_name:
+    		return conf.host_name
```

Comment:
```suggestion
		return conf.host_name
```

Reply:
Also, shouldn't the port get added here as well?

`frappe.utils.data.get_url` has a more elaborate implementation.

Reply:
According to [precedent](https://github.com/frappe/frappe/blob/b9993a303be95874f644c1f85d1e08a23be02509/.github/helper/db/mariadb.json#L15), `conf.host_name` is inclusive of the port...

This needs cleanup, just not sure how pervasive that cleanup would be.

Would you be happy with a follow up PR?

Reply:
@barredterra I rebased and othewise attended your recommendations. I'd like to refer this last change to a follow up PR, though as it seems to may have non-trivial side effect.

<sub> And while at it I also rebased https://github.com/frappe/frappe/pull/21975, https://github.com/frappe/frappe/pull/21874 & https://github.com/frappe/frappe/pull/21873 fixing more tests run in a production scenario. :smile: </sub>